num_enum_derive/src/lib.rs

```diff
@@ -248,7 +248,20 @@ impl Parse for EnumInfo {
 
             let mut next_discriminant = literal(0);
             for variant in data.variants.into_iter() {
+                let mut span = variant.span();
+                // Sadly `Span::join` is a nightly-only API at the moment - see https://github.com/rust-lang/rust/issues/54725
+                if let Some(span_with_fields) = span.join(variant.fields.span()) {
+                    span = span_with_fields;
+                }
```

Comment:
FWIW, if we were to use `Error::new_spanned(variant` over `Error::new(variant.span()` (the `die!` macro currently leads us to the latter; maybe we could try to feature a `die_spanned_at` variant for the former), then we'd have a correctly spanned error message, even on stable Rust. Indeed, `new_spanned` uses a hack to achieve this:

```rust
// imbued with the _start_ of the span
// vvvvvvvvvvvvvv
   compile_error! { "…" }
//                ^^^^^^^
//                imbued with the _end_ of the span
```

effectively yielding an error which does span over the whole range. Demo using `new_spanned(variant` _vs._ `new(variant.span()`:

<img width="472" alt="Screen Shot 2021-05-30 at 11 44 36" src="https://user-images.githubusercontent.com/9920355/120099563-6a446d80-c13c-11eb-847e-bfec39dbc89f.png">

Reply:
Ooh, TIL! This is amazing - done - thanks for the suggestion!